WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -145,12 +145,7 @@ private void changeLanguage(String languageCode) {
         conf.locale = newLocale;
         res.updateConfiguration(conf, res.getDisplayMetrics());
 
-        if (LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext()).equals(newLocale)) {
-            // remove custom locale key when original device locale is selected
-            mSettings.edit().remove(LANGUAGE_PREF_KEY).apply();
-        } else {
-            mSettings.edit().putString(LANGUAGE_PREF_KEY, newLocale.toString()).apply();
-        }
+        mSettings.edit().putString(LANGUAGE_PREF_KEY, newLocale.toString()).apply();
```

Comment:
> Actually, that flow might make more sense...

I agree, it makes sense to me that if the app is using the device language, the app language should change if the user updates the device language.